scripts/purge_messages.py

```diff
@@ -0,0 +1,33 @@
+import typer
+from prediction_market_agent_tooling.gtypes import private_key_type
+
+from prediction_market_agent.db.agent_communication import (
+    fetch_count_unprocessed_transactions,
+    pop_message,
+)
+from prediction_market_agent.utils import APIKeys
+
+
+def main(private_key: str) -> None:
+    keys = APIKeys(BET_FROM_PRIVATE_KEY=private_key_type(private_key))
+    n_messages = fetch_count_unprocessed_transactions(
+        consumer_address=keys.bet_from_address
+    )
+
+    if (
+        input(
+            f"Are you sure you want to purge all {n_messages} messages for agent {keys.bet_from_address}? (y/n): "
+        )
+        != "y"
+    ):
+        return
+
+    popped = 0
+    while fetch_count_unprocessed_transactions(consumer_address=keys.bet_from_address):
+        pop_message(api_keys=keys)
```

Comment:
I would have expected that `fetch_count_unprocessed_transactions` is called only once, yielding N unprocessed transactions. Afterwards, a for loop is created with exactly N, and those are then deleted.
Any reason for continuously sending RPC calls?

Reply:
In case it goes out of sync for some reason. But it's just script to be run "once in a decade", does that really matter?